modules/alerts/health-checks.tf

```diff
@@ -1,29 +1,33 @@
 locals {
 
   health_check_alerts = flatten([
-    for health_check in var.health_checks : [
-      {
-        name               = "${health_check.host}:${health_check.port}${health_check.path}-Main"
-        description        = "Main monitoring for ${health_check.host}"
-        source             = "AWS/Route53/HealthCheckStatus"
-        filters            = { Host = health_check.host }
-        statistic          = try(health_check.main.statistic, "min")
-        equation           = try(health_check.main.equation, "lt")
-        threshold          = try(health_check.main.threshold, 1)
-        period             = try(health_check.main.period, 60)
-        treat_missing_data = "breaching"
-      },
-      {
-        name               = "${health_check.host}:${health_check.port}${health_check.path}-Percentage"
-        description        = "Percentage monitoring for ${health_check.host}"
-        source             = "AWS/Route53/HealthCheckPercentageHealthy"
-        filters            = { Host = health_check.host }
-        statistic          = try(health_check.percentage.statistic, "avg")
-        equation           = try(health_check.percentage.equation, "lt")
-        threshold          = try(health_check.percentage.threshold, 75)
-        period             = try(health_check.percentage.period, 60)
-        treat_missing_data = "breaching"
-      }
+    for aws_hc in aws_route53_health_check.health_checks : [
+      for var_hc in local.health_checks :
+      aws_hc.fqdn == var_hc.host && tostring(aws_hc.port) == tostring(var_hc.port) && aws_hc.resource_path == var_hc.path ?
```

Comment:
why we need to double for constructions, with extra double loop on top of original one?
it seems you just need to get ws_route53_health_check.health_checks["${health_check.host}:${health_check.port}${health_check.path}"].id for each health check item so that you can leave original code and just change to the following in filters line:
```hcl
filters = { HealthCheckId = ws_route53_health_check.health_checks["${health_check.host}:${health_check.port}${health_check.path}"].id  }
```